source/fuzz/transformation_flatten_conditional_branch.cpp

```diff
@@ -843,85 +843,123 @@ void TransformationFlattenConditionalBranch::
-        // predecessor is associated with the "true" path of the conditional
-        // we are flattening, the other with the "false" path, but these
-        // predecessors can appear in either order as operands to the OpPhi
-        // instruction.  We determine in which order the OpPhi inputs should
-        // appear as OpSelect arguments by checking dominance of the true and
-        // false immediate successors of the header block.
-        if (ir_context->GetDominatorAnalysis(header_block.GetParent())
-                ->Dominates(branch_instruction_true_block_id,
-                            phi_inst->GetSingleWordInOperand(1))) {
-          // The "true" branch is handled first in the OpPhi's operands; we
-          // thus provide operands to OpSelect in the same order that they
-          // appear in the OpPhi.
-          operands.emplace_back(phi_inst->GetInOperand(0));
-          operands.emplace_back(phi_inst->GetInOperand(2));
-        } else {
-          // The "false" branch is handled first in the OpPhi's operands; we
-          // thus provide operands to OpSelect in reverse of the order that
-          // they appear in the OpPhi.
-          operands.emplace_back(phi_inst->GetInOperand(2));
-          operands.emplace_back(phi_inst->GetInOperand(0));
-        }
-        phi_inst->SetOpcode(SpvOpSelect);
-        phi_inst->SetInOperands(std::move(operands));
-      });
+          break;
+        case 3:
+          // Similar to the 2D case.
+          if (message_.fresh_id_for_bvec3_selector() != 0) {
+            selector_operand = {SPV_OPERAND_TYPE_ID,
+                                {message_.fresh_id_for_bvec3_selector()}};
+            require_3d_boolean_vector = true;
+          }
+          break;
+        case 4:
+          // Similar to the 2D case.
+          if (message_.fresh_id_for_bvec4_selector() != 0) {
+            selector_operand = {SPV_OPERAND_TYPE_ID,
+                                {message_.fresh_id_for_bvec4_selector()}};
+            require_4d_boolean_vector = true;
+          }
+          break;
+        default:
+          assert(dimension == 4 && "Invalid vector dimension.");
+          break;
+      }
+    }
+    std::vector<opt::Operand> operands;
+    operands.emplace_back(selector_operand);
+
+    uint32_t branch_instruction_true_block_id =
+        branch_instruction.GetSingleWordInOperand(1);
+    uint32_t branch_instruction_false_block_id =
+        branch_instruction.GetSingleWordInOperand(2);
+
+    // The OpPhi takes values from two distinct predecessors.  One
+    // predecessor is associated with the "true" path of the conditional
+    // we are flattening, the other with the "false" path, but these
+    // predecessors can appear in either order as operands to the OpPhi
+    // instruction.  We determine in which order the OpPhi inputs should
+    // appear as OpSelect arguments by checking dominance of the true and
+    // false immediate successors of the header block.
```

Comment:
Maybe update this comment slightly to include that we first check if either immediate successor is the convergence block.

Reply:
Done.